cloudinit/config/cc_wireguard.py

```diff
@@ -0,0 +1,242 @@
+            f' {wg_int["config_path"]}:{NL}{str(e)}'
+        ) from e
+
+
+def enable_wg(wg_int: dict, cloud: Cloud):
+    """Enable and start Wireguard interface
+
+    @param wg_int: Dict of configuration value under 'wg:interfaces'.
+
+    @raises: RuntimeError for issues enabling WG interface.
+    """
+    try:
+        LOG.debug("Enabling %s at boot", {wg_int["name"]})
+        cloud.distro.manage_service("enable", wg_int["name"])
+        LOG.debug("Bringing up interface %s", {wg_int["name"]})
+        cloud.distro.manage_service("start", wg_int["name"])
+    except subp.ProcessExecutionError as e:
+        raise RuntimeError(
+            f"Failed enabling/starting Wireguard interface(s):{NL}{str(e)}"
+        ) from e
+
+
+def readinessprobe_command_validation(wg_readinessprobes: list):
+    """Basic validation of user-provided probes
+
+    @param wg_readinessprobes: List of readinessprobe probe(s).
+
+    @raises: ValueError of wrong datatype provided for probes.
+    """
+    errors = []
+    pos = 0
+    for c in wg_readinessprobes:
+        if not isinstance(c, str):
+            errors.append(
+                f"Expected a string for readinessprobe at {pos}. Found {c}"
+            )
+            pos += 1
+
+    if errors:
+        raise ValueError(
+            f"Invalid readinessProbe commands:{NL}{NL.join(errors)}"
+        )
+
+
+def readinessprobe(wg_readinessprobes: list):
+    """Execute provided readiness probe(s)
+
+    @param wg_readinessprobes: List of readinessprobe probe(s).
+
+    @raises: ProcessExecutionError for issues during execution of probes.
+    """
+    errors = []
+    for c in wg_readinessprobes:
+        try:
+            LOG.debug("Running readinessprobe: '%s'", {str(c)})
+            subp.subp(c, capture=True, shell=True)
+        except subp.ProcessExecutionError as e:
+            errors.append(f"{c}: {e}")
+
+    if errors:
+        raise RuntimeError(
+            f"Failed running readinessprobe command:{NL}{NL.join(errors)}"
+        )
+
+
+def maybe_install_wireguard_tools(cloud: Cloud):
+    """Install wireguard-tools if not present."""
+    if subp.which("wg"):
+        return
+    try:
+        cloud.distro.update_package_sources()
+    except Exception:
+        util.logexc(LOG, "Package update failed")
+        raise
+    try:
+        cloud.distro.install_packages(["wireguard-tools"])
+    except Exception:
+        util.logexc(LOG, "Failed to install wireguard-tools")
+        raise
+    try:
+        subp.subp("modprobe wireguard", capture=True, shell=True)
+    except subp.ProcessExecutionError as e:
+        util.logexc(LOG, f"Could not load wireguard module:{NL}{str(e)}")
+        raise
+
+
+def handle(name: str, cfg: dict, cloud: Cloud, log, args: list):
+    wg_section = None
+
+    if "wireguard" in cfg:
+        LOG.debug("Found Wireguard section in config")
+        wg_section = cfg["wireguard"]
+
+    if wg_section is None:
```

Comment:
Why allow an empty config that just gets skipped? I don't understand that. Instead could we alter the schema to not disallow `null`, and just skip this module if the `wireguard` key doesn't exist in the config?

Reply:
Totally agree with you. I am going to implement it that way.